core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/protocol/json/JsonProtocolMarshaller.java

```diff
@@ -130,6 +125,24 @@ private static MarshallerRegistry createMarshallerRegistry() {
             .build();
     }
 
+    private static Map<MarshallLocation, TimestampFormatTrait.Format> getDefaultTimestampFormats() {
+        Map<MarshallLocation, TimestampFormatTrait.Format> formats = new HashMap<>();
+        formats.put(MarshallLocation.HEADER, TimestampFormatTrait.Format.ISO_8601);
```

Comment:
Header default format is rfc822

Reply:
Our protocol tests have ISO for marshalling and RFC822 for unmarshalling, not sure why. Did a search and no service is currently using timestamps in input headers so we may just have this wrong in our tests. I'll create a task to investigate.

Reply:
Yeah, for v1, we have to keep using iso as default for header marshalling for backwards compatibility reason. For v2, I think we can make it right and keep the same behavior as other SDKs.

Reply:
I think we can change it to Rfc822. Or you can add a TODO to investigate

Reply:
Add a TODO?